packages/trader/src/Stores/Modules/Trading/Helpers/chart.ts

```diff
@@ -0,0 +1,166 @@
+            },
+            event_type: chart_event_type,
+        };
+        if (state === STATE_TYPES.CHART_MODE_TOGGLE) {
+            payload.data.action = open_close_action;
+            return payload;
+        } else if (state === STATE_TYPES.CHART_TYPE_CHANGE) {
+            payload.data.action = 'choose_chart_type';
+            return payload;
+        } else if (state === STATE_TYPES.CHART_INTERVAL_CHANGE) {
+            payload.data.action = 'choose_time_interval';
+            return payload;
+        }
+    }
+    if (indicator_type_name && indicators_category_name) {
+        const payload = {
+            data: {
+                action: '',
+                indicator_type_name,
+                indicators_category_name,
+            },
+            event_type: indicators_event_type,
+        };
+        if (state === STATE_TYPES.INDICATOR_DELETED) {
+            payload.data.action = 'delete_active';
+            return payload;
+        } else if (state === STATE_TYPES.INDICATOR_SETTINGS_OPEN) {
+            payload.data.action = 'edit_active';
+            return payload;
+        } else if (state === STATE_TYPES.INDICATOR_INFO_TOGGLE) {
+            payload.data.action = is_info_open ? 'info_open' : 'info_close';
+            return payload;
+        }
+    }
+    if (state === STATE_TYPES.INDICATORS_MODAL_TOGGLE && 'is_open' in option) {
+        return {
+            data: { action: open_close_action },
+            event_type: indicators_event_type,
+        };
+    }
+    if (state === STATE_TYPES.INDICATOR_ADDED && indicator_type_name) {
+        return {
+            data: {
+                action: 'add_active',
+                indicator_type_name,
+                indicators_category_name,
+                subform_name: is_info_open ? 'indicators_info' : 'indicators_type',
+            },
+            event_type: indicators_event_type,
+        };
+    }
+    if (state === STATE_TYPES.INDICATORS_CLEAR_ALL) {
+        return {
+            data: { action: 'clean_all_active' },
+            event_type: indicators_event_type,
+        };
+    }
+    if (option.search_string && (state === STATE_TYPES.MARKET_SEARCH || state === STATE_TYPES.INDICATOR_SEARCH)) {
+        return {
+            data: {
+                action: 'search',
+                search_string,
+            },
+            event_type: state === STATE_TYPES.MARKET_SEARCH ? market_event_type : indicators_event_type,
+        };
+    }
+    if (state === STATE_TYPES.MARKETS_LIST_TOGGLE && 'is_open' in option) {
+        return {
+            data: {
+                action: open_close_action,
+                market_type_name,
+            },
+            event_type: market_event_type,
+        };
+    }
+    if (state === STATE_TYPES.SYMBOL_CHANGE && market_type_name) {
+        return {
+            data: {
+                action: 'choose_market_type',
+                market_type_name,
+                tab_market_name,
```

Comment:
i think we should rename this to instrument_type or instrument_name. According to https://deriv.com/markets/forex/ we are calling them instruments.

Reply:
@henry-deriv this is not what we choose, it's a request from marketing. Examples:
<img width="1317" alt="Screenshot 2023-09-06 at 10 57 22 AM" src="https://github.com/binary-com/deriv-app/assets/103177211/d9c2000b-16b7-4255-824c-bd1f45511fb4">

Reply:
@henry-deriv btw the template is updated: 
<img width="726" alt="Screenshot 2023-09-06 at 11 44 33 AM" src="https://github.com/binary-com/deriv-app/assets/103177211/485bcfa7-0b23-4172-9ec9-8beb735b047c">